src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1895,7 +1895,7 @@ private CSDiagnosticInfo NotFound(SyntaxNode where, string simpleName, int arity
             // as a type forwarder.  We'll look for type forwarders in the containing and
             // referenced assemblies and report more specific diagnostics if they are found.
             AssemblySymbol forwardedToAssembly;
-            bool encounteredForwardingCycle;
+            DiagnosticInfo encounteredForwardingError;
```

Comment:
This local feels suspicious.  It's declared at an outer scope but used as a distinct local in this scope and an inner scope.